sources/logdog/src/log_request.rs

```diff
@@ -205,12 +218,114 @@ where
     Ok(())
 }
 
+/// Copies all file matching the glob pattern given by `request.instructions` to the tempdir with filename and path
+/// same as source file.
+fn handle_glob_request<P>(request: &LogRequest<'_>, tempdir: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
+    })?;
+    for entry in glob_paths {
+        if let Ok(path) = entry {
+            if path.is_dir() {
+                // iterate the directory and copy files in the directory, nested
+                // sub-directories files will be copied only if matched by glob pattern.
+                for candidate in WalkDir::new(&path)
+                    .max_depth(1)
+                    .into_iter()
+                    .flat_map(|e| e.context(error::DirectoryWalk))
+                    .map(|e| e.into_path())
+                {
+                    if candidate.is_file() {
+                        file_copy(&request, &tempdir, &candidate)?;
+                    }
+                }
+            } else {
+                file_copy(&request, &tempdir, &path)?;
+            }
+        }
+    }
+    Ok(())
+}
+
+fn file_copy<P>(request: &LogRequest<'_>, destdir: P, src_filepath: &PathBuf) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    // with glob pattern there are chances of multiple targets with same name, therefore
+    // we maintain source file path and name in destination directory.
+    // Eg. src file path "/a/b/file" will be converted to "dest_dir/a/b/file"
+    let relative_path = &src_filepath
+        .strip_prefix("/")
+        .unwrap_or(src_filepath);
+    let dest_filepath = destdir.as_ref().join(relative_path);
+    let dest_dir_path = dest_filepath.parent().unwrap_or(&Path::new(""));
+    // create directories in dest file path if it does not exist
+    fs::create_dir_all(dest_dir_path).context(error::CreateOutputDirectory {
+        path: dest_dir_path,
+    })?;
+    let _ = fs::copy(&src_filepath, &dest_filepath).with_context(|| error::FileCopy {
+        request: request.to_string(),
+        from: src_filepath.to_str().unwrap_or_default(),
```

Comment:
```suggestion
        from: src_filepath.to_str().unwrap_or("<unknown>"),
```
Gives slightly more context in the error message, rather than having a blank.